src/governance/governance.cpp

```diff
@@ -746,19 +746,6 @@ MessageProcessingResult CGovernanceManager::SyncObjects(CNode& peer, CConnman& c
             continue;
         }
 
-        if (peer.nVersion < GOVSCRIPT_PROTO_VERSION && govobj.GetObjectType() == GovernanceObject::PROPOSAL) {
```

Comment:
> Support for P2SH proposals was introduced in Dash Core 0.18, we don't
> have to worry about antagonizing those peers as they're multiple hard
> forks behind us.

That's true for Dash Core. But beside Dash Core there could be 3rd party apps who support old p2p version only.
I would not just remove it from p2p code.

Maybe it's better to return some error, but not quietly ignoring version.

Reply:
@UdjinM6, thoughts?

Reply:
I would generally say, if `GOVSCRIPT_PROTO_VERSION` is greater than the minimum P2P protocol version, then we should keep this logic.